gems/aws-sdk-core/lib/aws-sdk-core/instance_credentials_provider.rb

```diff
@@ -22,16 +22,24 @@ class InstanceCredentialsProvider < Hearth::IdentityProvider
     # Initializes an instance of InstanceCredentialsProvider using ENV and
     # profile values.
     def self.from_env(config)
+      return if ENV['AWS_EC2_METADATA_DISABLED']
       profile = config[:profile]
       profile_config = AWS::SDK::Core.shared_config.profiles[profile]
-      return if ENV['AWS_EC2_METADATA_DISABLED']
 
       endpoint = ENV['AWS_EC2_METADATA_SERVICE_ENDPOINT'] ||
-                 profile_config['ec2_metadata_service_endpoint']
+                 (profile_config &&
+                   profile_config['ec2_metadata_service_endpoint'])
       endpoint_mode = ENV['AWS_EC2_METADATA_SERVICE_ENDPOINT_MODE'] ||
-                      profile_config['ec2_metadata_service_endpoint_mode']
+                 (profile_config &&
+                   profile_config['ec2_metadata_service_endpoint_mode'])
       client = EC2Metadata.new(endpoint: endpoint, endpoint_mode: endpoint_mode)
+      # temporary - possible fix
```

Comment:
Whats the issue here?

Reply:
This was for whether instance creds should be always resolved last or a possible no cred providers at all